### PR TITLE
ref(auth): Use Feature component to gate auth providers

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Exposes Auth to organizations that don't have SSO features, and gates each provider behind it's required feature.

tunnel evanpurkhiser com_settings_evanpurkhiser_auth_ 2

⚠️Tests still need updated.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_auth_providers.py [hunk 1]
```diff
@@ -22,14 +21,11 @@ def get(self, request, organization):
...
-            is_saml = issubclass(v, SAML2Provider)
-            if is_saml and not HAS_SAML2:
-                continue
-
-            feature = v.required_feature
-            if feature and not features.has(feature, organization, actor=request.user):
-                continue
-
-            provider_list.append((k, v.name, is_saml))
+            provider_list.append({
```
Comment (billyvg) [inline src/sentry/api/endpoints/organization_auth_providers.py]: Is this endpoint documented/public? Did we check to see if any other views are using it?

**Review by billyvg — COMMENTED** [review]
frontend lgtm

File: src/sentry/api/endpoints/organization_auth_providers.py (hunk 1 above)
Comment (EvanPurkhiser) [reply src/sentry/api/endpoints/organization_auth_providers.py]: This API isn't documented no, and this is the only view using it.

File: src/sentry/static/sentry/app/views/settings/organizationAuth/providerItem.jsx [hunk 2]
```diff
@@ -1,16 +1,21 @@
...
+import Hovercard from 'app/components/hovercard';
+import SentryTypes from 'app/sentryTypes';
+import Tag from 'app/views/settings/components/tag';
 
 export default class ProviderItem extends React.PureComponent {
   static propTypes = {
-    providerKey: PropTypes.string.isRequired,
-    providerName: PropTypes.string.isRequired,
+    organization: SentryTypes.Organization.isRequired,
+    provider: PropTypes.object.isRequired,
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/organizationAuth/providerItem.jsx]: Yeah sounds good, will add.

File: src/sentry/static/sentry/app/views/settings/organization/navigationConfiguration.jsx [hunk 3]
```diff
@@ -39,8 +39,7 @@ const organizationNavigation = [
       {
         path: `${pathPrefix}/auth/`,
         title: t('Auth'),
-        show: ({organization, access, features}) =>
-          features.has('sso-basic') && access.has('org:admin'),
+        show: ({organization, access, features}) => access.has('org:admin'),
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/organization/navigationConfiguration.jsx]: Do you need to unpack `organization` and `features` if you don't use them? :bike:

**Review by markstory — APPROVED** [review]
Looks good to me.

File: src/sentry/static/sentry/app/views/settings/organization/navigationConfiguration.jsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/organization/navigationConfiguration.jsx]: nope

**Pull request merged**